common/libs/units/units.h

```diff
@@ -115,12 +115,38 @@ template <class Q, class ValTy> class ArithScalar : public Scalar<Q, ValTy> {
 public:
   constexpr Q operator+(const ArithScalar &a) { return Q(this->val_ + a.val_); }
   constexpr Q operator-(const ArithScalar &a) { return Q(this->val_ - a.val_); }
+  inline ArithScalar &operator+=(const ArithScalar &a) {
+    return *this = *this + a;
+  }
+  inline ArithScalar &operator-=(const ArithScalar &a) {
+    return *this = *this - a;
+  }
 
 protected:
   // Pull in base class's constructor.
   using Scalar<Q, ValTy>::Scalar;
 };
 
+// Represents an ArithScalar where multiplication by an external factor of type
+// ValTy is defined, creating what mathematicians call a "module":
+// https://en.wikipedia.org/wiki/Module_(mathematics)
+// Multiplication by a ValTy factor follows the rules of multiplication for
+// ValTy, and requires multiplication to be well defined on ValTy type, except
+// that it does not commute: float*ModuleScalar is undefined
+// This is especially useful for things that need averaging.
+// Be aware that dividing by 0.0f yields Inf and needs to be protected.
+template <class Q, class ValTy>
+class ModuleScalar : public ArithScalar<Q, ValTy> {
+public:
+  constexpr Q operator*(const ValTy &a) { return Q(this->val_ * a); }
+  constexpr Q operator/(const ValTy &a) { return Q(this->val_ / a); }
+  inline ModuleScalar &operator*=(const ValTy &a) { return *this = *this * a; }
+  inline ModuleScalar &operator/=(const ValTy &a) { return *this = *this / a; }
```

Comment:
Any reason for ArithScalar not to have *, /, *=, and /=?  Like, do we have units which shouldn't be modules?

Reply:
Multiplying pressure feels a bit weird to me as it is a relative value and multiplying it makes little sense (except maybe for averaging). The others could probably all be modules.

Reply:
OK, I'd be down for that (maybe with a comment on Pressure saying what you just said)?

Reply:
Done